### PR TITLE
udp-broadcast-relay-redux: Add package

### PR DESCRIPTION
**Pull request by accwebs**
Add package for udp-broadcast-relay-redux (https://github.com/udp-redux/udp-broadcast-relay-redux). 

Description: This program listens for packets on a specified UDP broadcast port.  When a packet is received, it sends that packet to all specified interfaces but the one it came from as though it originated from the original sender. The primary purpose of this is to allow games on machines on separated local networks (Ethernet, WLAN) that use udp broadcasts to find each other to do so.
It also works on ppp links, so you can log in from windows boxes (e.g. using pptp) and play LAN-based games together. Currently, you have to care about upcoming or downgoing interfaces yourself.

My interest: I found this simple utility to be pretty helpful in forwarding UDP broadcast traffic across two networks joined by tun-based VPN (tun does not support broadcast packets so use --target override on src and dest).

Other info: This commit is largely based on the work from Daniel Dickinson in PR #2096 which was never merged. I tweaked it in a number of ways. All bugs with this package are mine, not his ;-)

Maintainer: me
Compile tested: mvebu, Linksys wrt1200ac, bleeding edge
Compile tested: mvebu, Linksys wrt1200ac, 19.07.7
Run tested: mvebu, Linksys wrt1200ac, bleeding edge, verified binary at least launches
Run tested: mvebu, Linksys wrt1200ac, 19.07.7, verified traffic forwarding works in my home network config

Signed-off-by: Aaron Curley <accwebs@gmail.com>

Caution:  Noob PR. I don't quite know what I'm doing to apologies in advance for clueless questions.  :-) 
Also, I listed myself as maintainer above but didn't put myself in the Makefile.  I can *try* to act as maintainer from this point forward but I don't entirely know what I'm doing.

### CODE REVIEW

File: net/udp-broadcast-relay-redux-openwrt/Makefile
Comment (accwebs) [inline net/udp-broadcast-relay-redux-openwrt/Makefile:11]: Is there some more appropriate version than 1.0.0?  The underlying utility doesn't include a version number.

File: net/udp-broadcast-relay-redux-openwrt/Makefile
Comment (accwebs) [inline net/udp-broadcast-relay-redux-openwrt/Makefile:20]: I got this value by running the build and then running sha256sum against the tarball in the 'dl' dir.  Hopefully that's right?

File: net/udp-broadcast-relay-redux-openwrt/files/udp_broadcast_relay_redux.config
Comment (accwebs) [inline net/udp-broadcast-relay-redux-openwrt/files/udp_broadcast_relay_redux.config:1]: should this filename use dashes instead of underscores?

**Comment by accwebs** [comment]
Do people think adding this package (in its current limited state) is valuable?  If not, I won't be offended.

  E.g., given the lack of proper hooks for interface status changes?

File: net/udp-broadcast-relay-redux-openwrt/Makefile
Comment (1715173329) [reply net/udp-broadcast-relay-redux-openwrt/Makefile:11]:
```suggestion
```
Okay, just drop this line.
It will be filled automatically.

See also:
https://github.com/openwrt/openwrt/blob/f6cc00650cbdc887a6b7557e6934dec3a04e31da/include/download.mk#L13
```
PKG_VERSION := $(if $(PKG_SOURCE_DATE),$(PKG_SOURCE_DATE)-)$(call version_abbrev,$(PKG_SOURCE_VERSION))
```

File: net/udp-broadcast-relay-redux-openwrt/Makefile
Comment (1715173329) [inline net/udp-broadcast-relay-redux-openwrt/Makefile:2]: Please use your own Copyright, or leave it blank.

File: net/udp-broadcast-relay-redux-openwrt/Makefile
Comment (1715173329) [inline net/udp-broadcast-relay-redux-openwrt/Makefile:17]:
```suggestion
```
If not **necessary**, just drop this, as it will be filled automatically.

File: net/udp-broadcast-relay-redux-openwrt/Makefile
Comment (1715173329) [inline net/udp-broadcast-relay-redux-openwrt/Makefile:18]: Please add PKG_SOURCE_DATE, like this:
```suggestion
PKG_SOURCE_DATE:=2021-04-05
PKG_SOURCE_VERSION:=671372938b55a186625a80516f86e8b9948c977a
```

File: net/udp-broadcast-relay-redux-openwrt/Makefile
Comment (1715173329) [reply net/udp-broadcast-relay-redux-openwrt/Makefile:20]: It's okay.

Another way is setting `PKG_MIRROR_HASH` to `skip`, then run `make package/<your package name>/check FIXUP=1`

File: net/udp-broadcast-relay-redux-openwrt/Makefile
Comment (1715173329) [inline net/udp-broadcast-relay-redux-openwrt/Makefile:19]:
```suggestion
```
Same reason like `PKG_SOURCE_SUBDIR`.

File: net/udp-broadcast-relay-redux-openwrt/files/udp_broadcast_relay_redux.config
Comment (1715173329) [reply net/udp-broadcast-relay-redux-openwrt/files/udp_broadcast_relay_redux.config:1]: Both names are okay :-)
But... Why do you comment out all of the config?

File: net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init
Comment (1715173329) [inline net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init:75]:
```suggestion
    procd_add_reload_trigger "udp_broadcast_relay_redux"
```
This should match the config name.

File: net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init
Comment (1715173329) [inline net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init:2]: Please use your own Copyright, or leave it blank.

File: net/udp-broadcast-relay-redux-openwrt/Makefile
Comment (1715173329) [inline net/udp-broadcast-relay-redux-openwrt/Makefile:12]:
```suggestion
PKG_RELEASE:=$(AUTORELEASE)
```
Just use `$(AUTORELEASE)` for simplicity.
And calling `$(PKG_SOURCE_VERSION)` here won't work, as it isn't defined before referred.

**Comment by 1715173329** [comment]
Please short the commit title as well.
`udp-broadcast-relay-redux: add package` is okay.
And please only leave the **most important description** in commit message.

Thanks for your hard work.

**Comment by accwebs** [comment]
> Please short the commit title as well.
> `udp-broadcast-relay-redux: add package` is okay.
> And please only leave the **most important description** in commit message.
> 
> Thanks for your hard work.

Hi @1715173329, 

Thanks so much for the review -- your detailed feedback is exactly what I needed!  I know reviewing code can be tedious so I DO appreciate your time here.

I don't have time today, but I'll take care of these items tomorrow most likely.

File: net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init
Comment (accwebs) [reply net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init:75]: > Why do you comment out all of the config?

I wondered about that as well (to comment it out or not).  To me, it didn't seem like my example config was ever likely to be desired by anyone without them first tweaking it for their use case (specific computer game or other use case).  Hence why I felt commenting it out was better.

But, if you think it's more sane to leave it un-commented, I'm also fine with that!  Admittedly I'm kinda unsure about all this stuff so I'll definitely defer to you!

File: net/udp-broadcast-relay-redux-openwrt/Makefile
Comment (accwebs) [reply net/udp-broadcast-relay-redux-openwrt/Makefile:18]: Will do!  My guess is that should always match the most recent commit date?

**Comment by accwebs** [comment]
@1715173329 questions about the appropriate content for the commit message.  If I trim it down to:

```
udp-broadcast-relay-redux: Add package

This commit is largely based on the work from Daniel Dickinson in PR #2096 which was never merged. I tweaked it in a number of ways. All bugs with this package are mine, not his

Maintainer: me
Compile tested: mvebu, Linksys wrt1200ac, bleeding edge
Compile tested: mvebu, Linksys wrt1200ac, 19.07.7
Run tested: mvebu, Linksys wrt1200ac, bleeding edge, verified binary at least launches
Run tested: mvebu, Linksys wrt1200ac, 19.07.7, verified traffic forwarding works in my home network config

Signed-off-by: <Aaron Curley accwebs@gmail.com>
```

Is that right?  Or:

1. should I be eliminating the maintainer, compile tested, runtime tested, etc parts?
2. should I remove the attribution to the original author (it felt like attribution was appropriate since I based my work on his)

File: net/udp-broadcast-relay-redux-openwrt/Makefile
Comment (1715173329) [reply net/udp-broadcast-relay-redux-openwrt/Makefile:18]: No, it should match the `PKG_SOURCE_VERSION` commit date.

File: net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init
Comment (1715173329) [reply net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init:75]: > Hence why I felt commenting it out was better.

Oh I see, and thanks for explanation.
But please at least keep `config udp_broadcast_relay_redux` uncommented.

**Comment by 1715173329** [comment]
Hi @accwebs,

> 1. should I be eliminating the maintainer, compile tested, runtime tested, etc parts?

Yes, please remove them in your commit message.

> 2. should I remove the to the original author (it felt like attribution was appropriate since I based my work on his)

You should keep it.

And please warp a new line every 70 characters, in your commit message.

File: net/udp-broadcast-relay-redux-openwrt/Makefile
Comment (accwebs) [reply net/udp-broadcast-relay-redux-openwrt/Makefile:18]: Yeah, that's what I meant.  Got it!

File: net/udp-broadcast-relay-redux-openwrt/Makefile
Comment (accwebs) [reply net/udp-broadcast-relay-redux-openwrt/Makefile:20]: :+1:

File: net/udp-broadcast-relay-redux-openwrt/Makefile
Comment (accwebs) [reply net/udp-broadcast-relay-redux-openwrt/Makefile]: Fixed.  Also removed from the init script.

**Comment by 1715173329** [comment]
Looks much better now!
Thanks for your hard work.
Please remember to squash all your changes into a single commit :-)

**Comment by accwebs** [comment]
I just did but I still gotta work on the config file.  Give me a bit.  I want to test that the init script doesn't choke on the config section with commented-out options.

File: net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init
Comment (accwebs) [reply net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init]: Ok, is that expected to work (config section that's empty)?

Comment (1715173329) [reply net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init]: It's okay.

Comment (accwebs) [reply net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init]: Hmm, I just tested it with the following config:

```
config udp_broadcast_relay_redux
#       option id 1
#       option port 47624
#       list network lan
#       list network vpnsrv
#       list network vpnsuperior
#       option dest_override 255.255.255.255
```

With this config:

```
root@wrt1200ac-Ocqueoc:~# /etc/init.d/udp-broadcast-relay-redux start
Validation failed
```

Whereas if it's entirely commented out 

```
/etc/init.d/udp-broadcast-relay-redux start
```
Does nothing.  Which is preferred?

File: net/udp-broadcast-relay-redux-openwrt/Makefile
Comment (1715173329) [reply net/udp-broadcast-relay-redux-openwrt/Makefile:11]: I see CI reports the correct HASH is `11cf8728f2b8e966f4f57032d817a889f680ed8e61afff35b52ca9c6789a03c6`.
Could you please have a check?

File: net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init
Comment (1715173329) [reply net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init]: In my option, doing nothing is better :-)

**Comment by accwebs** [comment]
Checking on the package hash.

**Comment by accwebs** [comment]
Ok, I updated the package hash.  I don't know how I got that wrong.  Hopefully it's correct now?  Or maybe my edits caused it to change.

**Comment by 1715173329** [comment]
Great, it's correct now.

**Review by 1715173329 — APPROVED** [review]

**Comment by 1715173329** [comment]
Kindly ping @neheb.
Could you please make a review?

**Comment by accwebs** [comment]
Thanks for all your help.  Also it is too late in the day now but in the next few days I'll attempt to re-test the latest package on my router.

**Comment by 1715173329** [comment]
It's my pleasure :-)
Good night~

File: net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init
Comment (neheb) [inline net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init:53]: Replace with -n

File: net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init
Comment (neheb) [inline net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init:57]: Replace with -n

File: net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init
Comment (accwebs) [reply net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init]: Done!

File: net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init
Comment (neheb) [inline net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init:37]: Shellcheck report https://github.com/koalaman/shellcheck/wiki/SC2004

**Comment by accwebs** [comment]
Hah, I know it's already merged but I *did* just finish re-testing the latest package on my router and it seems ok!

**Comment by accwebs** [comment]
Thanks all for your help on this :-)

**Pull request merged**
